Unity-Technologies/Graphics
# Fixed NRE exception

### Purpose of this PR
https://fogbugz.unity3d.com/f/cases/1389076/

How to reproduce:
1. Open the attached VFX Graph and drag it to the Scene
2. Attach it to the GameObject and Select Init context
3. Observe that Gizmos are invisible due to Arc Cone's space being set to World

👍  **Expected result**: Gizmos are visible when changing spaces of the block
👎  **Actual result**: Gizmos are invisible and errors are thrown to the console

Reproduced with: 2021.2.4f1 (12.1.1) , 2022.1.0b2(13.1.3)

Error message:

> NullReferenceException: Object reference not set to an instance of an object
> UnityEditor.VFX.UI.VFXDataAnchorGizmoContext+<>c__DisplayClass11_0.<BuildValue>b__3 (System.Collections.Generic.List`1[T] o) (at Library/PackageCache/com.unity.visualeffectgraph@13.1.3/Editor/GraphView/Elements/Controllers/VFXDataAnchorController.cs:742)

[Before]
![Screenshot 2021-12-16 at 15 03 42](https://user-images.githubusercontent.com/85612173/147102628-82d340e2-5537-4774-a63f-76d0494b38f7.png)

[After]
<img width="1653" alt="Unity_d8qSfp7hbM" src="https://user-images.githubusercontent.com/85612173/147104369-aa3a2750-450b-4610-a57d-49995e36b4b2.png">

---
### Testing status
Tested the scenario described above

---
### ⚠ Comments to reviewers
This PR does not really fix the issue because two different bugs are behind this issue.
1- There's an exception raised in the VFX code which prevents to run further instructions
2- There's an issue (a warning message is displayed in the console) with quaternion preventing the gizmo from being displayed
Here is the console message:
> Look rotation viewing vector is zero

The current PR fixes the first issue. The second one seems to be fixed in trunk (2022.2-a01).
So with this PR it should be completely fixed in 2022.2.

To backport the fix we need to backport this PR off course, but we also need to identify the PR that fixed the second issue to backport it too.
Currently the lead we found for this is this [bug](https://fogbugz.unity3d.com/f/cases/1284682/). The symptoms are similar but it seems to be related to Slider1D when in our case it's Slider2D.
